src/fileutil.c

```diff
@@ -63,6 +63,15 @@ rs_file_open(char const *filename, char const *mode)
 	    return stdin;
     }
 
+    if (!force && is_write) {
+	if ((f = fopen(filename, "rb"))) {
```

Comment:
"Rather than look-before-you-leap, it'd be better to open the file in a way that won't overwrite existing files, probably using open and then fdopen."

Using fdopen to switch to write mode 'do not cause truncation of the file.' FOPEN(3).
Closing and reopening the file in write mode seems the simplest solution ?